astropy/io/misc/yaml.py

```diff
@@ -223,6 +243,23 @@ def ignore_aliases(self, data):
 AstropyDumper.add_representer(TimeDelta, _timedelta_representer)
 AstropyDumper.add_representer(coords.SkyCoord, _skycoord_representer)
 
+# Numpy dtypes
+AstropyDumper.add_representer(np.bool_,
+                              yaml.representer.SafeRepresenter.represent_bool)
+for np_type in [np.int_, np.int, np.intc, np.intp, np.int8, np.int16, np.int32,
+                np.int64, np.uint8, np.uint16, np.uint32, np.uint64]:
+    AstropyDumper.add_representer(np_type,
+                                 yaml.representer.SafeRepresenter.represent_int)
+for np_type in [np.float_, np.float, np.float16, np.float32, np.float64,
+                np.longdouble]:
```

Comment:
longdouble is represented differently in different platform, but should be there. complex128 is just float 64 for real and imaginary, so it should be too.
For the loss of precision, I guess it will always be an issue when we save to text, whatever the representation, unless we save a lot of significant digits (which anyway does not work out of the box with a simple `print`) or split the number into two parts. I don't have a prefered solution.